streams/src/main/java/org/apache/kafka/streams/state/DslWindowParams.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.state;
+
+import java.time.Duration;
+import java.util.Objects;
+import org.apache.kafka.streams.kstream.EmitStrategy;
+
+/**
+ * {@code DslWindowParams} is a wrapper class for all parameters that function
+ * as inputs to {@link DslStoreSuppliers#windowStore(DslWindowParams)}.
+ */
+public class DslWindowParams {
+
+    private final String name;
+    private final Duration retentionPeriod;
+    private final Duration windowSize;
+    private final boolean retainDuplicates;
+    private final EmitStrategy emitStrategy;
+    private final boolean isSlidingWindow;
+
+    /**
+     * @param name             name of the store (cannot be {@code null})
+     * @param retentionPeriod  length of time to retain data in the store (cannot be negative)
+     *                         (note that the retention period must be at least long enough to contain the
+     *                         windowed data's entire life cycle, from window-start through window-end,
+     *                         and for the entire grace period)
```

Comment:
this is kind of a roundabout way of saying it must be at least equal to the windowSize + gracePeriod. I think it's better to just explicitly state this and put the constraint in concrete terms relative to other known quantities

Reply:
Actually IIUC users should never have to instantiate these classes directly, so I guess it's not that critical to define the specific constraints and ok to describe it conceptually. Or both 🤷‍♀️

Reply:
this was copied from javadoc elsewhere - see `Stores#persistentWindowStore`. if we want to change it we can do that in a separate PR so it's consistent throughout the board

Reply:
Ah, that's ok then, I thought you wrote this just for this class